docs/design/datacontracts/Loader.md

```diff
@@ -0,0 +1,110 @@
+# Contract Loader
+
+This contract is for getting information about loaded modules and assemblies
+
+## APIs of contract
+
+``` csharp
+readonly struct ModuleHandle
+{
+    // Opaque handle - no public members
+
+    internal TargetPointer Address;
+}
+
+[Flags]
+enum ModuleFlags
+{
+    EditAndContinue = 0x00000008,   // Edit and Continue is enabled for this module
+    ReflectionEmit = 0x00000040,    // Reflection.Emit was used to create this module
+}
+
+enum ModuleLookupTable
+{
+    FieldDefToDesc,
+    ManifestModuleReferences,
+    MemberRefToDesc,
+    MethodDefToDesc,
+    TypeDefToMethodTable,
+    TypeRefToMethodTable,
+}
```

Comment:
Since this is a contract, should these be explicitly defined? Is this part of the abstract API contract or the binary data description in the runtime?